src/test/unit_tests/test_s3_bucket_policy.js

```diff
@@ -860,4 +860,90 @@ mocha.describe('s3_bucket_policy', function() {
        assert.strictEqual(Object.keys(actualObjectTagCondition.StringEquals)[0], "s3:ExistingObjectTag/key");
        assert.strictEqual(actualObjectTagCondition.StringEquals["s3:ExistingObjectTag/key"], "value");
     });
+
+    mocha.it('should be able to use notAction', async function() {
+        const auth_put_policy = {
+        Version: '2012-10-17',
+        Statement: [
+            {
+                Effect: 'Allow',
+                Principal: { AWS: "*" },
+                NotAction: ['s3:GetObject'],
+                Resource: [`arn:aws:s3:::${BKT}/*`]
+            }
+        ]};
+        await s3_owner.putBucketPolicy({
+            Bucket: BKT,
+            Policy: JSON.stringify(auth_put_policy)
+        }).promise();
+
+        await s3_a.putObject({
+            Body: 'Some data for the file... bla bla bla... ',
```

Comment:
If you can, maybe use it as a constant variable (instead of writing the string every time).